shell/browser/api/electron_api_tray.cc

```diff
@@ -63,7 +63,7 @@ gin::WrapperInfo Tray::kWrapperInfo = {gin::kEmbedderNativeGin};
 Tray::Tray(gin::Handle<NativeImage> image,
            base::Optional<UUID> guid,
            gin::Arguments* args)
-    : tray_icon_(TrayIcon::Create(guid)) {
+    : tray_icon_(TrayIcon::Create(guid)), isolate_(args->isolate()) {
```

Comment:
there's no need to save isolates in the main process because there's only ever 1. it's `v8::Isolate::GetCurrent()`.

Reply:
@nornagon calling `v8::isolate::GetCurrent()` repeatedly is much more costly, since there are atomic loads in v8::isolate::GetCurrent(), so it's preferable to avoid doing so

Reply:
I don't think it makes a difference on the scale of this API. But if we're seriously worried about the performance overhead of the atomic load there, we should create a singleton that holds the isolate reference. It's a bit silly to pass it around everywhere.